dspace-api/src/main/java/org/dspace/uclouvain/itemEnhancer/poller/UCLouvainItemEnhancerUpdatePoller.java

```diff
@@ -0,0 +1,277 @@
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataFieldName;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.itemEnhancer.UCLouvainItemEnhancerService;
+import org.dspace.uclouvain.itemEnhancer.enhancers.ItemEnhancerConfiguration;
+import org.dspace.uclouvain.itemEnhancer.model.ItemToEnhance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+/**
+ * Poller to process desired items and update their metadata values.
+ * This class reads the 'uclouvain_item_authority_metadata_enhancement' table and process all its entries.
+ * Each time entries are pulled, the table is cleaned, and later refilled with other entries.
+ * One entry represent a link between two items: a 'target' and a 'source'.
+ * The main goal of this poller is to update the target item using the source item metadata
+ * and a specific configuration.
+ * 
+ * @author Michaël Pourbaix <michael.pourbaix@uclouvain.be>
+ */
+@Component
+@EnableScheduling
+public class UCLouvainItemEnhancerUpdatePoller {
+
+    @Autowired
+    UCLouvainItemEnhancerService itemEnhancerService;
+
+    @Autowired
+    ItemService itemService;
+
+    private Logger logger = LogManager.getLogger(UCLouvainItemEnhancerUpdatePoller.class);
+    private boolean isScheduledEnabled = DSpaceServicesFactory
+        .getInstance()
+        .getConfigurationService()
+        .getBooleanProperty("uclouvain-item-enhancer-poller.enabled", false);
+
+    /**
+     * Main poller task, executed every x seconds.
+     * NOTE: use 'fixedDelay' instead of 'fixedRate' since 'fixedDelay' waits for the execution to end
+     * before triggering countdown.
+     */
+    @Scheduled(fixedDelayString = "${uclouvain-item-enhancer-poller.delay}")
+    public void triggerCycleCheck() {
+        if (isScheduledEnabled) {
+            checkForItemUpdate();
+        }
+    }
+
+    /**
+     * Main method to trigger the main poller functionality.
+     * This method is called through triggerCycleCheck() with a configured fixed delay.
+     * For tests this method is called directly, this is why it is public.
+     */
+    public void checkForItemUpdate() {
+        Context context = new Context();
+        context.turnOffAuthorisationSystem();
+        try {
+            // Retrieve items to update form the database table.
+            List<ItemToEnhance> itemsToUpdate = itemEnhancerService.retrieveAllItemsToUpdate(context);
```

Comment:
You should use a limit, not return "allItems". If the table contains 1M of items to get, you could reach the "outofmemory" exception or have a method running for a lllllllong time and consuming a lot of resource. Maybe 1000 records is a suffisent limit. In best world, this limit could be defined into the bean and/or config file
PS : also change the name method ;-)